keras_nlp/models/bart/bart_seq_2_seq_lm_preprocessor.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""BART Seq2Seq LM preprocessor layer."""
+
+from absl import logging
+
+from keras_nlp.api_export import keras_nlp_export
+from keras_nlp.models.bart.bart_preprocessor import BartPreprocessor
+from keras_nlp.utils.keras_utils import pack_x_y_sample_weight
+
+
+@keras_nlp_export("keras_nlp.models.BartSeq2SeqLMPreprocessor")
+class BartSeq2SeqLMPreprocessor(BartPreprocessor):
+    """BART Seq2Seq LM preprocessor.
+
+    This layer is used as preprocessor for seq2seq tasks using the BART model.
+    This class subclasses `keras_nlp.models.BartPreprocessor` and keeps most of
+    its functionality. It has two changes from the superclass:
+
+     1. Sets the `y` (label) and `sample_weights` fields by shifting the
+        decoder input sequence one step towards the left. Both these fields are
+        inferred internally, and any passed values will be ignored.
+     2. Drops the last token from the decoder input sequence as it does not have
+        a successor.
+
+    Args:
+        tokenizer: A `keras_nlp.models.BartTokenizer` instance.
+        encoder_sequence_length: The length of the packed encoder inputs.
+        decoder_sequence_length: The length of the packed decoder inputs.
+        truncate: string. The algorithm to truncate a list of batched segments
+            to fit within `sequence_length`. The value can be either
+            `round_robin` or `waterfall`:
+                - `"round_robin"`: Available space is assigned one token at a
+                    time in a round-robin fashion to the inputs that still need
+                    some, until the limit is reached.
+                - `"waterfall"`: The allocation of the budget is done using a
+                    "waterfall" algorithm that allocates quota in a
+                    left-to-right manner and fills up the buckets until we run
+                    out of budget. It supports an arbitrary number of segments.
+
+    Call arguments:
+        x: A dictionary with `encoder_text` and `decoder_text` as its keys.
+            Each value in the dictionary should be a tensor of single string
+            sequences. Inputs may be batched or unbatched. Raw python inputs
+            will be converted to tensors.
+        y: Any label data. Any passed value will be ignored since this is
```

Comment:
Copy the language here https://github.com/keras-team/keras-nlp/blob/df2e85ebb4c99afdf58655c56e574f492fa519ce/keras_nlp/models/bert/bert_masked_lm_preprocessor.py#L71-L73